hcloud/action.go

```diff
@@ -125,3 +125,62 @@ func (c *ActionClient) All(ctx context.Context) ([]*Action, error) {
 
 	return allActions, nil
 }
+
+// Wait watches the action state until it completes with success or error.
+// Also provides progress updates on the first channel.
+func (c *ActionClient) Wait(ctx context.Context, action *Action) (<-chan int, <-chan error) {
+	errCh := make(chan error, 1)
+	progressCh := make(chan int)
+
+	go func() {
+		defer close(errCh)
+		defer close(progressCh)
+
+		ticker := time.NewTicker(100 * time.Millisecond)
+		sendProgress := func(p int) {
+			select {
+			case progressCh <- p:
+				break
+			default:
+				break
+			}
+		}
+
+		var retries = 0
```

Comment:
What about resetting it to 0 after a successful response?

Reply:
done